File: Modulo.Net/aula06/Crescer.PetStore/src/PetStore.Dominio/Entidades/Categoria.cs
namespace PetStore.Dominio.Entidades
{
    public class Categoria
    {
        private Categoria() { }

        public Categoria(string nome)
        {
            this.Nome = nome;
        }

        public int Id { get; set; }

        public string Nome { get; private set; }
    }
}

File: ModuloJava/aula02/LocadoraVHS/src/main/java/dominio/Categoria.java
package dominio;

public enum Categoria {
    DOURADA(1),
    PRATA(2),
    BRONZE(3);

    private int categoria;

    Categoria(int categoria) {
        this.categoria = categoria;
    }

    public double getValor() {
        switch (categoria) {
            case 1:
                return 10;
            case 2:
                return 7;
            case 3:
                return 3;
            default:
                return -1;
        }
    }

    public int getPrazo() {
        switch (categoria) {
            case 1:
                return 2;
            case 2:
                return 3;
            case 3:
                return 5;
            default:
                return -1;
        }
    }
}
